WordPress/Classes/Services/MenusService.m

```diff
@@ -161,47 +159,21 @@ - (void)deleteMenu:(Menu *)menu
                     failure:failure];
 }
 
-- (void)generateDefaultMenuItemsForBlog:(Blog *)blog
-                                success:(nullable void(^)(NSArray <MenuItem *> * _Nullable defaultItems))success
-                                failure:(nullable MenusServiceFailureBlock)failure
-{
-    // Get the latest list of Pages available to the site.
-    PostServiceSyncOptions *options = [[PostServiceSyncOptions alloc] init];
-    options.statuses = @[PostStatusPublish];
-    options.order = PostServiceResultsOrderAscending;
-    options.orderBy = PostServiceResultsOrderingByTitle;
-    PostService *postService = [[PostService alloc] initWithManagedObjectContext:self.managedObjectContext];
-    [postService syncPostsOfType:PostServiceTypePage
-                     withOptions:options
-                         forBlog:blog
-                         success:^(NSArray *pages) {
-                             [self.managedObjectContext performBlock:^{
-                                 
-                                 if (!pages.count) {
-                                     success(nil);
-                                     return;
-                                 }
-                                 NSMutableArray *items = [NSMutableArray arrayWithCapacity:pages.count];
-                                 // Create menu items for the top parent pages.
-                                 for (Page *page in pages) {
-                                     if ([page.parentID integerValue] > 0) {
-                                         continue;
-                                     }
-                                     MenuItem *pageItem = [NSEntityDescription insertNewObjectForEntityForName:[MenuItem entityName] inManagedObjectContext:self.managedObjectContext];
-                                     pageItem.contentID = page.postID;
-                                     pageItem.name = page.titleForDisplay;
-                                     pageItem.type = MenuItemTypePage;
-                                     [items addObject:pageItem];
-                                 }
-                                 
-                                 [[ContextManager sharedInstance] saveContext:self.managedObjectContext withCompletionBlock:^{
-                                     if (success) {
-                                         success(items);
-                                     }
-                                 } onQueue:dispatch_get_main_queue()];
-                             }];
-                         }
-                         failure:failure];
+- (MenuItem *)createItemWithPageID:(NSManagedObjectID *)pageObjectID inContext:(NSManagedObjectContext *)context {
+    Page *page = [context existingObjectWithID:pageObjectID error:nil];
```

Comment:
Have you considered passing an error object?

Reply:
That didn't cross my mind. That's a good idea though: turning this Objective-C function to a throwing Swift function.

But unfortunately, after giving it go, I noticed Swift turns the function to `createItem() throws -> MenuItem`, but I want an optional return value: `createItem() throws -> MenuItem?`. I think I'll keep it as it is, because returning an non optional MenuItem is a lie.